client/lib/media/actions.js

```diff
@@ -212,29 +218,28 @@ MediaActions.add = function( siteId, files ) {
 	}, Promise.resolve() );
 };
 
-MediaActions.edit = function( siteId, item ) {
-	var newItem = assign( {}, MediaStore.get( siteId, item.ID ), item );
+MediaActions.cleanTemporaryData = function( siteId, item, fields ) {
+	for ( let i = 0; i < fields.length; i++ ) {
+		delete item[ fields[ i ] ];
+	}
 
-	Dispatcher.handleViewAction( {
+	const updateAction = {
 		type: 'RECEIVE_MEDIA_ITEM',
-		siteId: siteId,
-		data: newItem
-	} );
-};
+		siteId,
+		data: item
+	};
 
-MediaActions.update = function( siteId, item, editMediaFile = false ) {
-	if ( Array.isArray( item ) ) {
-		item.forEach( MediaActions.update.bind( null, siteId ) );
-		return;
-	}
+	Dispatcher.handleViewAction( updateAction );
+};
 
+MediaActions.updateTemporally = function( type, siteId, item ) {
```

Comment:
Let's add some tests to cover this action, new behavior in MediaActions.update, and the 1.1 vs 1.2 api version switch